In [7]:
install.packages("readxl")
library(readxl)
df <- read_excel("/star digital.xls")
str(df)
head(df)

data_csv <- read.csv("/star_digital.csv")
head(data_csv)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



tibble [25,303 × 9] (S3: tbl_df/tbl/data.frame)
 $ id      : num [1:25303] 545716 893524 1372718 971359 59999 ...
 $ purchase: num [1:25303] 1 1 1 1 1 1 0 0 1 0 ...
 $ test    : num [1:25303] 1 1 1 1 1 0 1 0 1 1 ...
 $ imp_1   : num [1:25303] 0 1 0 14 0 0 2 0 97 0 ...
 $ imp_2   : num [1:25303] 1 0 0 37 0 1 272 0 214 0 ...
 $ imp_3   : num [1:25303] 0 0 0 1 0 0 0 0 0 0 ...
 $ imp_4   : num [1:25303] 0 17 10 7 13 0 18 0 11 0 ...
 $ imp_5   : num [1:25303] 0 0 0 0 0 0 0 0 3 0 ...
 $ imp_6   : num [1:25303] 0 1 0 7 0 0 2 1 13 2 ...


id      purchase test imp_1 imp_2 imp_3 imp_4 imp_5 imp_6
1  545716 1        1     0     1    0      0    0     0    
2  893524 1        1     1     0    0     17    0     1    
3 1372718 1        1     0     0    0     10    0     0    
4  971359 1        1    14    37    1      7    0     7    
5   59999 1        1     0     0    0     13    0     0    
6  842034 1        0     0     1    0      0    0     0

id      purchase test imp_1 imp_2 imp_3 imp_4 imp_5 imp_6
1  545716 1        1     0     1    0      0    0     0    
2  893524 1        1     1     0    0     17    0     1    
3 1372718 1        1     0     0    0     10    0     0    
4  971359 1        1    14    37    1      7    0     7    
5   59999 1        1     0     0    0     13    0     0    
6  842034 1        0     0     1    0      0    0     0

In [14]:
#1. Is online advertising effective for Star Digital? In other words, is there a difference in conversion rate between the treatment and control groups?
library(dplyr)

logit_model <- glm(purchase ~ test, data = df, family = binomial)

summary(logit_model)


Call:
glm(formula = purchase ~ test, family = binomial, data = df)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept) -0.05724    0.03882  -1.474   0.1404  
test         0.07676    0.04104   1.871   0.0614 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 35077  on 25302  degrees of freedom
Residual deviance: 35073  on 25301  degrees of freedom
AIC: 35077

Number of Fisher Scoring iterations: 3


The online advertising is not effective for Star Digital, since the p-value > 0.05 (p-value=0.0614), the null hypothesis cannot be rejected, which means there is no statistically significant evidence whether advertising can increase purchase rates. The impact of advertising on purchases may be due to random factors. Although the regression coefficients indicate that the ad group has a higher probability of purchase, this difference may be due to random fluctuations and is not statistically significant.

In [20]:
# 2. Is there a frequency effect of advertising on purchase? In particular, the question is whether increasing the frequency of advertising (number of impressions) increases the probability of purchase?

df$total_impressions <- rowSums(df[, c("imp_1", "imp_2", "imp_3", "imp_4", "imp_5", "imp_6")])

logit_freq_model <- glm(purchase ~ total_impressions + test, data = df, family = binomial)

summary(logit_freq_model)



Call:
glm(formula = purchase ~ total_impressions + test, family = binomial, 
    data = df)

Coefficients:
                   Estimate Std. Error z value Pr(>|z|)    
(Intercept)       -0.244063   0.040023  -6.098 1.07e-09 ***
total_impressions  0.029180   0.001294  22.551  < 2e-16 ***
test               0.072376   0.041697   1.736   0.0826 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 35077  on 25302  degrees of freedom
Residual deviance: 34209  on 25300  degrees of freedom
AIC: 34215

Number of Fisher Scoring iterations: 5


The number of ad impressions has a significant impact on purchases: the the coefficient of total_impressions is positive and the p-value is very small, indicating that the more times an ad is shown, the more likely a user is to purchase.
The effect of the ad group on purchases is marginally significant: the the p-value for the test variable is close to 0.1, which suggests that a difference between the ad group and the control group may exist, but there is not enough evidence that the effect of the ad group is more significant than the control group.

In [24]:
# 3. How does the conversion effectiveness of Sites 1-5 compare with that of Site 6?

df$site_group <- ifelse(df$test == 1, "Sites_1_5", "Site_6") # Changed 'data' to 'df'

logit_site_model <- glm(purchase ~ site_group + total_impressions, data = df, family = binomial) # Changed 'data' to 'df'

summary(logit_site_model)


Call:
glm(formula = purchase ~ site_group + total_impressions, family = binomial, 
    data = df)

Coefficients:
                     Estimate Std. Error z value Pr(>|z|)    
(Intercept)         -0.244063   0.040023  -6.098 1.07e-09 ***
site_groupSites_1_5  0.072376   0.041697   1.736   0.0826 .  
total_impressions    0.029180   0.001294  22.551  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 35077  on 25302  degrees of freedom
Residual deviance: 34209  on 25300  degrees of freedom
AIC: 34215

Number of Fisher Scoring iterations: 5


Number of ad impressions significantly affects purchase: total_impressions has a significant impact on purchase, with each additional ad impression increasing the log odds of purchase.
The effect of site grouping is marginally significant: site_group's effect on purchases approaches significance at the 0.1 level, but does not quite reach the standard significance level of 0.05. Although Sites_1_5 has a slight increase in the log odds of purchase compared to Sites_6, the difference is not strong.

In [25]:
# 4. Optional Challenge Question -- Which sites should Star Digital advertise on? In particular, should it put its advertising dollars in Site 6 or in Sites 1 through 5?
summary(logit_site_model)


Call:
glm(formula = purchase ~ site_group + total_impressions, family = binomial, 
    data = df)

Coefficients:
                     Estimate Std. Error z value Pr(>|z|)    
(Intercept)         -0.244063   0.040023  -6.098 1.07e-09 ***
site_groupSites_1_5  0.072376   0.041697   1.736   0.0826 .  
total_impressions    0.029180   0.001294  22.551  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 35077  on 25302  degrees of freedom
Residual deviance: 34209  on 25300  degrees of freedom
AIC: 34215

Number of Fisher Scoring iterations: 5


According to the model results: Advertising strategy: From the regression analysis, although the impact of sites 1-5 is slightly advantageous compared to site 6, you can consider investing more advertising budget in sites 1-5 because the ad conversion rate of these sites is slightly higher than that of site 6.
The impact of ad impressions: No matter which site it is placed on, increasing the number of ad impressions will significantly increase the chance of purchase, so increasing the frequency of ad display is very important to improve conversion rates.
Recommendation: Prioritize advertising on Sites 1-5, but also consider whether Site 6 has a specific target user or market.
Increase the number of ad impressions, whether you choose Sites 1-5 or Site 6, more ad impressions can effectively increase conversion rates.